executor.go

```diff
@@ -902,6 +902,38 @@ type FieldResolver interface {
 	Resolve(p ResolveParams) (interface{}, error)
 }
 
+func defaultResolveStruct(sourceVal reflect.Value, fieldName string) (interface{}, error) {
+	for i := 0; i < sourceVal.NumField(); i++ {
+		valueField := sourceVal.Field(i)
+		typeField := sourceVal.Type().Field(i)
+		// try matching the field name first
+		if strings.EqualFold(typeField.Name, fieldName) {
+			return valueField.Interface(), nil
+		}
+		if typeField.Anonymous && typeField.Type.Kind() == reflect.Struct {
+			return defaultResolveStruct(valueField, fieldName)
```

Comment:
There's a bug here.  We return too early if an anonymous struct comes before a field we're looking for.

I think what we want to do is run through all the top-level fields first, then dive if the field wasn't found.
Repeat for all anonymous structs until it's found.  If not found, then return `nil, nil`.

Reply:
Here's a fix, sans a proper test:

https://github.com/abraithwaite/graphql/commit/5decd5b2db5eb8145c6f79927a7b2341a3780ce5

Reply:
Good catch, @abraithwaite